### PR TITLE
Scripts: Add check-engines script to the package

### PR DESCRIPTION
**Pull request by gziolo**
## Description
This PR adds `check-engines` to `wp-scripts` exposed by `@wordpress/scripts` package.

It checks if the current `node`, `npm` (or `yarn`) versions match the given semver version ranges. If the given version is not satisfied, information about installing the needed version is printed and the program exits with an error code. It uses [check-node-version](https://www.npmjs.com/package/check-node-version) behind the scenes with the default configuration provided. You can specify your own ranges as described in [check-node-version docs](https://www.npmjs.com/package/check-node-version).

## How has this been tested?
* `npm run check-engines` still works properly.
* `./node_modules/.bin/wp-scripts check-engines --node ">=11"` - shows error message
* `./node_modules/.bin/wp-scripts check-engines --npm ">=7"` - shows error message

### CODE REVIEW

File: packages/scripts/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 2.5.0 (Unreleased)
+
+### New Feature
+
+- Added support for `check-engines` script ([#XXXXX](https://github.com/WordPress/gutenberg/pull/XXXXX))
```
Comment (adamsilverstein) [inline packages/scripts/CHANGELOG.md]:
```suggestion
- Added support for `check-engines` script ([#12721](https://github.com/WordPress/gutenberg/pull/12721))
```

**Review by adamsilverstein — APPROVED** [review]
Overall looks good and verified command still works as expected. One note: after running `npm install` my package-lock.json file was changed (https://cl.ly/40878777f402).

File: packages/scripts/README.md [hunk 2]
```diff
@@ -27,6 +28,24 @@ _Example:_
 
 ## Available Scripts
 
+### `check-engines`
+
+Check if the current `node`, `npm` (or `yarn`) versions match the given semver version ranges. If the given version is not satisfied, information about installing the needed version is printed and the program exits with an error code. It uses [check-node-version](https://www.npmjs.com/package/check-node-version) behind the scenes with the default configuration provided. You can specify your own ranges as described in [check-node-version docs](https://www.npmjs.com/package/check-node-version).
```
Comment (ntwb) [inline packages/scripts/README.md]:
```suggestion
Check if the current `node`, `npm` (or `yarn`) versions match the given [semantic version](https://semver.org/) ranges. If the given version is not satisfied, information about installing the needed version is printed and the program exits with an error code. It uses [check-node-version](https://www.npmjs.com/package/check-node-version) behind the scenes with the default configuration provided. You can specify your own ranges as described in [check-node-version docs](https://www.npmjs.com/package/check-node-version).
```

Comment (ntwb) [reply packages/scripts/README.md]: Added link to semver.org

File: packages/scripts/scripts/check-engines.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const args = getCliArgs();
+
+const hasConfig = hasCliArg( '--package' ) ||
+	hasCliArg( '--node' ) ||
+	hasCliArg( '--npm' ) ||
+	hasCliArg( '--yarn' );
+const config = ! hasConfig ?
+	[
+		'--node', '>=8.0.0',
```
Comment (ntwb) [inline packages/scripts/scripts/check-engines.js]:
```suggestion
		'--node', '>=10.0.0',
```

As _core_ has switched to using Node.js 10.x we should also use 10.x here

**Review by ntwb — APPROVED** [review]
Looks good, added a couple of minor changes

File: packages/scripts/scripts/check-engines.js (hunk 3 above)
Comment (gziolo) [reply packages/scripts/scripts/check-engines.js]: I wasn't sure. Sounds awesome 👍
Was it announced somewhere? It would be nice to just merge it :)

**Review by ntwb — APPROVED** [review]
:shipit:

**Pull request merged**